rclpy/src/rclpy_common/include/rclpy_common/exceptions.hpp

```diff
@@ -79,6 +79,12 @@ class UnsupportedEventTypeError : public RCLError
 {
   using RCLError::RCLError;
 };
+
+class NotImplementedError : public RCLError
```

Comment:
@gbalke hmm, does `RCL_RET_UNSUPPORTED` set an `rcl` error?

Reply:
I assumed it was in the same boat as `NodeNameNonExistant` https://github.com/ros2/rcl/blob/0ca8c6162136594e490ba69fbf288de1b5f7073d/rcl/src/rcl/common.c